docs/configuration/network.md

```diff
@@ -42,6 +42,7 @@ license: |
 | celeborn.&lt;role&gt;.rpc.dispatcher.threads | &lt;value of celeborn.rpc.dispatcher.threads&gt; | false | Threads number of message dispatcher event loop for roles |  |  | 
 | celeborn.io.maxDefaultNettyThreads | 64 | false | Max default netty threads | 0.3.2 |  | 
 | celeborn.network.bind.preferIpAddress | true | false | When `true`, prefer to use IP address, otherwise FQDN. This configuration only takes effects when the bind hostname is not set explicitly, in such case, Celeborn will find the first non-loopback address to bind. | 0.3.0 |  | 
+| celeborn.network.bind.wildcardAddress | false | false | When `true`, the bind address will be set to a wildcard address, while the advertise address will remain as whatever is set by `celeborn.network.bind.preferIpAddress`. This is helpful in dual-stack environments, where the service must listen to both IPv4 and IPv6 clients. | 0.6.0 |  | 
```

Comment:
These two configurations can be still confusing: celeborn.network.bind.wildcardAddress and celeborn.network.bind.preferIpAddress. When the celeborn.network.bind.wildcardAddress setting is active, the server will listen on a wildcard address (in this case, celeborn.network.bind.preferIpAddress will not be active). However, celeborn.network.bind.preferIpAddress is still active in other scenarios. In my opinion, we should distinguish between the concepts of "to bind" and "to listen" for these configurations.

Reply:
What do you suggest @RexXiong ? Do you have any naming recommendations? 
cc @SteNicholas @waitinfuture @pan3793 @FMX

Reply:
>  In my opinion, we should distinguish between the concepts of "to bind" and "to listen" for these configurations.

`bind` and `listen` are the same concept, if I understand correctly, @akpatnam25 you are requesting a configurable `advertiseAddress`

Reply:
@pan3793 no this PR just configures allows RPC and HTTP servers to bind to wildcard address. 
Initially, when `celeborn.network.bind.preferIpAddress` is set, the bind address and advertise address are both determined by `celeborn.network.bind.preferIpAddress` (either ip address or hostname for both). Now, if  `celeborn.network.bind.wildcardAddress` is set, advertise address will still be either ip address or hostname based on `celeborn.network.bind.preferIpAddress`, but bind address will be wildcard address. 

hope this clarifies :)

Reply:
How about using celeborn.network.bind.advertiseAddress.preferIp to replace celeborn.network.bind.preferIpAddress and making it fall back to celeborn.network.bind.preferIpAddress? @akpatnam25  cc @pan3793

Reply:
Sure, but I think that is still misleading, given that `celeborn.network.bind.advertiseAddress.preferIp` can still be used for binding AND advertising when `celeborn.network.bind.wildcardAddress` is set to false.

Reply:
Essentially, these are the 2 scenarios: 
1. `celeborn.network.bind.wildcardAddress=true`
    - RPC and HTTP servers bind to wildcard address
    - Advertise address (either hostname or ip) is determined by `celeborn.network.bind.preferIpAddress`
2. `celeborn.network.bind.wildcardAddress=false`
    - RPC and HTTP servers bind to the address (either hostname or ip) determined by `celeborn.network.bind.preferIpAddress`.
    - Advertise address either hostname or ip) determined by `celeborn.network.bind.preferIpAddress`.

Reply:
I suppose we can say that in the case that `celeborn.network.bind.wildcardAddress=true`, advertise address can be determined by `celeborn.network.bind.advertiseAddress.preferIp`?  Otherwise, if `celeborn.network.bind.wildcardAddress=false`, then bind is determined by `celeborn.network.bind.preferIpAddress` and advertise is determined by `celeborn.network.bind.advertiseAddress.preferIp`. What do you think @RexXiong @pan3793 ?

Reply:
The specific scenario where this becomes relevant is as follows:

If some of the compute infrastructure is IPv4 only, some v6 only and others dual stack - the way we can have a single Celeborn infra to cater to all is by:
a) Set prefer ip to false - so that advertised address is the host and not IP.

b) bind to wild card address

With both in place, the v4 only will resolve the v4 address and connect to v4 ip/port.
Likewise, for v6 only.
Dual stack compute nodes will use prefer ipv6 Java flag to resolve to either v4 or v6.

Reply:
> I suppose we can say that in the case that `celeborn.network.bind.wildcardAddress=true`, advertise address can be determined by `celeborn.network.bind.advertiseAddress.preferIp`? Otherwise, if `celeborn.network.bind.wildcardAddress=false`, then bind is determined by `celeborn.network.bind.preferIpAddress` and advertise is determined by `celeborn.network.bind.advertiseAddress.preferIp`. What do you think @RexXiong @pan3793 ?

I'm okay with this. For compatibility, I suggest that celeborn.network.bind.advertiseAddress.preferIp should default to the value of celeborn.network.bind.preferIpAddress. WDYT?

Reply:
sounds good, I can make these changes later today.